extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -462,6 +464,31 @@ public ThreadFactory createThreadFactory() {
         };
     }
 
+    public ContextHandler<Object> executionContextHandler() {
+        return new ContextHandler<Object>() {
+            @Override
+            public Object captureContext() {
+                return Vertx.currentContext();
+            }
+
+            @Override
+            public void runWith(Runnable task, Object context) {
+                if (context != null) {
+                    // Only do context handling if it's non null
+                    final ContextInternal vertxContext = (ContextInternal) context;
+                    try {
+                        vertxContext.beginDispatch();
```

Comment:
```suggestion
                    vertxContext.beginDispatch();
                    try {
```

I'd do it like this, because you don't want `endDispatch` to run if `beginDispatch` failed for some reason.